drivers/sensor/tmp112/tmp112.c

```diff
@@ -48,38 +47,39 @@ static int tmp112_reg_read(struct tmp112_data *drv_data,
 	return 0;
 }
 
-static int tmp112_reg_write(struct tmp112_data *drv_data,
+static int tmp112_reg_write(const struct device *dev,
 			    uint8_t reg, uint16_t val)
 {
+	const struct tmp112_config *config = dev->config;
+
 	uint16_t val_be = sys_cpu_to_be16(val);
 
-	return i2c_burst_write(drv_data->i2c, TMP112_I2C_ADDRESS,
+	return i2c_burst_write(config->bus, config->i2c_slv_addr,
 			       reg, (uint8_t *)&val_be, 2);
 }
 
-static int tmp112_reg_update(struct tmp112_data *drv_data, uint8_t reg,
+static int tmp112_reg_update(const struct device *dev, uint8_t reg,
 			     uint16_t mask, uint16_t val)
 {
 	uint16_t old_val = 0U;
 	uint16_t new_val;
 
-	if (tmp112_reg_read(drv_data, reg, &old_val) < 0) {
+	if (tmp112_reg_read(dev, reg, &old_val) < 0) {
 		return -EIO;
 	}
 
 	new_val = old_val & ~mask;
 	new_val |= val & mask;
 
-	return tmp112_reg_write(drv_data, reg, new_val);
+	return tmp112_reg_write(dev, reg, new_val);
 }
 
 static int tmp112_attr_set(const struct device *dev,
 			   enum sensor_channel chan,
 			   enum sensor_attribute attr,
 			   const struct sensor_value *val)
 {
-	struct tmp112_data *drv_data = dev->data;
-	int64_t value;
+	uint16_t value;
```

Comment:
Is changing the type related to adding multi-instance support? If not, please make this change in a separate commit.

Reply:
Moved to a separate commit